app/main.py

```diff
@@ -2,5 +2,19 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def inner(*args) -> str:
+        if func.__name__ not in cached_results:
+            print("Calculating new result")
+            cached_results[func.__name__] = {args: func(*args)}
+        else:
+            if args in cached_results[func.__name__]:
+                print("Getting from cache")
+            else:
+                print("Calculating new result")
+                cached_results[func.__name__][args] = func(*args)
```

Comment:
This is a clean code issue. You are printing messages directly in the code. This is fine for debugging, but you should remove these print statements in the final version of the code. If you want to keep some logging, consider using the `logging` module, which gives you more control over the output.